tests/unit/test_config.py

```diff
@@ -0,0 +1,84 @@
+from copy import deepcopy
+
+import pytest
+from pydantic.error_wrappers import ValidationError
+
+from bugjira.config import Config
+
+
+def test_config_good_json(config_defaults):
+    """
+    GIVEN a json file containing a valid definition of a Bugjira config dict
+    WHEN we call Config.from_config with its pathname
+    THEN no exception is raised
+    """
+    config = Config.from_config(
+        config_path=(config_defaults + "/data/config/good_config.json")
+    )
+    assert config is not None
+
+
+def test_config_good_dict(good_config_dict):
+    """
+    GIVEN a well-defined dict containing a Bugjira config
+    WHEN we call Config.from_config using the dict as the config_dict
+    THEN no exception is raised
+    AND the dict returned is the dict that was passed in
+    """
+    assert Config.from_config(config_dict=good_config_dict) == good_config_dict
+
+
+def test_config_missing_values(config_defaults):
+    """
+    GIVEN a json file containing an incomplete definition of a Bugjira config
+    WHEN we call Config.from_config with the path to the file
+    THEN a ValidationError is raised
+    """
+    with pytest.raises(ValidationError):
```

Comment:
While this test is ok as-is, if you add 'as error' to your with statement here, you can then assert what the error is, specifically. This would also provide the detail @jpichon was looking for, I think, with regard to what exactly the failed validation will return.

Reply:
ACK, I am good with that!